### ISSUE TITLE
Stratford Theme: Button Text Color Not Changing From Button Block Setting

### ISSUE THREAD

**Issue by Codexpart** [Automattic/themes#2844]
## Steps to replicate

1. Activate the Stratford Theme.
2. Add a blank page.
3. Add dummy Content with Heading and Links
4. Add Button and change color form listed colors on Text Color and update the page
5. Color not changed
6. But If we select from Custom color picker. it works perfectly

Page Editor:
![2020-11-22_16-30-04](https://user-images.githubusercontent.com/7956548/99901550-968cb680-2ce1-11eb-957b-cdd612b9e0b1.png)
After Publish:
![2020-11-22_16-32-05](https://user-images.githubusercontent.com/7956548/99901555-a3a9a580-2ce1-11eb-8e58-c215784b9f5e.png)

<h2>Browser / OS version</h2>

<p>Google Chrome — Version 86.0.4240.111 (Official Build) (64-bit)<br>Windows 10 Home Single Language — 20H2 (64-bit).</p>

## Result

Buttons text color not changed after choosing colors from age Editor > Button Block Setting > Color settings > Text Color

## Expected

Change the Color of Buttons Text, When choosing colors from Page Editor > Button Block Setting > Color settings > Text Color

* Follow-up ticket: 3509871-zd

**Comment by eduardozulian** [comment]
On top of that, looks like the `Outline` style is not reflected in the editor, sometimes only when you visit the published post/page.

**Comment by kriskarkoski** [comment]
Another report in 26272178-hc. Custom color workaround worked.

**Comment by mdrockwell** [comment]
Another report here: 26275152-hc

**Comment by mikeicode** [comment]
Ran into this on 4544682-hc

Testing this I see there are multiple CSS issues. 

**Issue 1**

If you have a background color set on a surrounding block (such as the Group block this demo adds for this section), this class `has-background` is added to that block which results in the following CSS affecting the color of the button text:

```
.has-background a {
    color: currentColor;
}
```

![https://d.pr/i/YAOa1w](https://d.pr/i/YAOa1w+)

**Issue 2**

The class `has-secondary-background-color` may be added to the button depending on the button background color you choose. Then this CSS affects the button color:

```
.has-secondary-background-color {
    color: #fff;
}

```




**Temporary Solution**

It appears choosing a **Custom color** for either the text color or the button background color overrides these 2 issues since the color CSS is then added inline.

![https://d.pr/i/zPdZev](https://d.pr/i/zPdZev+)

**Comment by scruffian** [comment]
Fixed by #3185
